invenio_rdm_records/resources/serializers.py

```diff
@@ -9,32 +9,94 @@
 """Record response serializers."""
 
 import json
+from functools import partial
 
 from flask_resources.serializers import JSONSerializer
 
+from invenio_rdm_records.vocabularies import Vocabularies
+
 
 class UIJSONSerializer(JSONSerializer):
     """UI JSON serializer implementation."""
 
     def _serialize_access_right(self, obj):
-        """Dump ui config for `access_right` field."""
+        """Inject ui config for `access_right` field."""
         return dict(
             access_right=dict(
                 category=obj["access"]["access_right"]
             ))
 
+    def _serialize_ui_options_from_vocabulary(
+            self, vocabulary_name, top_bucket_key=None):
+        """Creates UI facet label options from RDM vocabularies.
+
+        :params top_bucket_key: Key used for nested buckets to indicate
+                                the top level aggregation.
+        """
+        ui_options = {}
+        vocabulary = Vocabularies.get_vocabulary(
+            vocabulary_name).dump_options()
+        if type(vocabulary) is dict:
+            ui_options[top_bucket_key] = {}
+            for key in vocabulary.keys():
+                for option in vocabulary[key]:
+                    ui_options[top_bucket_key][option['value']] = \
+                        str(option['text'])
+        else:
+            for option in vocabulary:
+                ui_options[option['value']] = str(option['text'])
+        return ui_options
+
+    def _set_bucket_label(self, labels_map, bucket):
+        """Set bucket label from config.
+
+        :params labels_map: Dict of transforming bucket values to labels.
+        :params bucket: Current aggregation bucket.
+        """
+        bucket_label = labels_map.get(bucket["key"])
+        if bucket_label:
+            bucket["label"] = bucket_label
```

Comment:
Minor: We should add the `doc_count` to the `label` here to keep the same visuals.